src/Microsoft.AspNetCore.Server.IntegrationTesting/Common/PortHelper.cs

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.AspNetCore.Server.IntegrationTesting
+{
+    public class PortHelper
+    {
+        public static void LogPortStatus(ILogger logger, int port)
+        {
+            logger.LogInformation("Checking for processes currently using port {0}", port);
+
+            var psi = new ProcessStartInfo
+            {
+                RedirectStandardOutput = true,
+                RedirectStandardError = true,
+            };
+
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                psi.FileName = "cmd";
+                psi.Arguments = $"/C netstat -nq | find \"{port}\"";
+            }
+            else
+            {
+                psi.FileName = "lsof";
+                psi.Arguments = $"-i :{port}";
+            }
+
+            var process = new Process
+            {
+                StartInfo = psi,
+                EnableRaisingEvents = true
+            };
+
+            var linesLogged = false;
+
+            process.OutputDataReceived += (sender, data) =>
+            {
+                if (!string.IsNullOrWhiteSpace(data.Data))
+                {
+                    linesLogged = true;
+                    logger.LogInformation("portstatus: {0}", data.Data);
+                }
+            };
+            process.ErrorDataReceived += (sender, data) =>
+            {
+                if (!string.IsNullOrWhiteSpace(data.Data))
+                {
+                    logger.LogWarning("portstatus: {0}", data.Data);
```

Comment:
stderr would indicate the lsof or netstat command failed to list port info or isn't available on the path.